messaging/write/index.d.ts

```diff
@@ -8,12 +8,13 @@ type CreateWriterOptions = {
   messageStore: MessageStore;
 };
 
-type WriteFn = (message: Message, streamName: string, options?: { expectedVersion?: number }) => Promise<number>;
+interface WriteFn {
+  (message: Message, streamName: string, options?: { expectedVersion?: number }): Promise<number>;
+}
 
-export type Writer = {
-  write: WriteFn;
+export interface Writer extends WriteFn {
   emitter: ClassLike;
   initial: (message: Message, streamName: string) => Promise<number>;
```

Comment:
❓ Looking over the template component, would each Handler have a `writeEvent` function that internally calls `write.initial` for each handled Message?  I feel like I'm missing something because it seems weird to call a method named `initial` more than once